moabb/datasets/Dreyer2023.py

```diff
@@ -0,0 +1,307 @@
+        ==========  =======  =======  ==========  =================  ============  ===============  ===========
+
+    ===================
+    Dataset description
+    ===================
+    A large EEG database with users' profile information for motor imagery
+    Brain-Computer Interface research
+
+    Data collectors : Appriou Aurélien; Caselli Damien; Benaroch Camille;
+                      Yamamoto Sayu Maria; Roc Aline; Lotte Fabien;
+                      Dreyer Pauline; Pillette Léa
+    Data manager : Dreyer Pauline
+    Project leader : Lotte Fabien
+    Project members : Rimbert Sébastien; Monseigne Thibaut
+
+    """
+
+    def __init__(self, db_id="A", subjects=[]):
+        assert db_id in [
+            "A",
+            "B",
+            "C",
+        ], "Invalid dataset selection! Existing Dreyer2023 datasets: A, B, and C."
+        self.db_id = db_id
+        self.db_idx_off = dict(A=0, B=60, C=81)
+
+        super().__init__(
+            subjects,
+            sessions_per_subject=1,
+            events=dict(left_hand=1, right_hand=2),
+            code="Dreyer2023",
+            interval=[3, 8],
+            paradigm="imagery",
+            doi="10.5281/zenodo.7554429",
+        )
+
+    def get_subject_info(self, subjects=[], infos=["Demo_Bio"]):
+        path = get_dataset_path("DREYER", path=None)
+        dlpath = dl.data_dl(DREYER2023_URL, "DREYER_2023", path)
+        perform_path = join(dirname(dlpath), "BCI Database", "Perfomances.xlsx")
+
+        df = pd.read_excel(perform_path)
+
+        if self.db_id == "A":
+            df.columns = df.iloc[1, :]
+            df = df.iloc[list(range(2, 62)), :]
+        if self.db_id == "B":
+            df.columns = df.iloc[65]
+            df = df.iloc[list(range(66, 87)), :]
+        if self.db_id == "C":
+            df.columns = df.iloc[90]
+            df = df.iloc[list(range(91, 97)), :]
+        df.reset_index(drop=True, inplace=True)
+        df.columns.name = None
+        if len(subjects):
+            subjects = [
+                self.db_id + str(s) if not str(s).startswith(self.db_id) else str(s)
+                for s in subjects
+            ]
+            assert not any(
+                [s for s in subjects if s not in df["SUJ_ID"].tolist()]
+            ), "Invalid subject selection."
+            df = df.loc[df["SUJ_ID"].isin(subjects)]
+
+        info_select = ["SUJ_ID"]
+        for i in infos:
+            if i in SUBJ_INFO.keys():
+                for j in SUBJ_INFO[i]:
+                    if j in df.columns:
+                        info_select.append(j)
+            elif i in df.columns:
+                info_select.append(i)
+            else:
+                raise "Ivalid info selection."
+        return df[info_select]
+
+    def _get_single_subject_data(self, subject):
+        subj_dir = self.data_path(subject)
+
+        subj_id = self.db_id + str(subject + self.db_idx_off[self.db_id])
+
+        ch_names = [
```

Comment:
To avoid long list like that you could use #fmt: off/on like [explained here](https://stackoverflow.com/a/58584557)